lib/ClimaCoreTempestRemap/src/netcdf.jl

```diff
@@ -0,0 +1,257 @@
+
+import ClimaCore: slab, column
+
+"""
+    def_time_coord(nc::NCDataset, length=Inf, eltype=Float64;
+        units = "seconds since 2020-01-01 00:00:00"
+        kwargs...
+    )
+
+Deine a time coordinate (dimension + variable) `"time"` in the NetCDF dataset
+`nc`. By default its length is set to be unlimited. The variable corresponding
+to the coordinate is returned.
+
+Additional attributes can be added as keyword arguments.
+
+# Example
+
+```julia
+timevar = add_time_coord!(nc; units = "seconds since 2020-01-01 00:00:00",)
+timevar[:] = collect(0.0:0.5:60)
+```
+"""
+function def_time_coord(
+    nc::NCDataset,
+    length = Inf,
+    eltype = Float64;
+    standard_name = "time",
+    long_name = "time",
+    axis = "T",
+    kwargs...,
+)
+    defDim(nc, "time", length)
+    time = defVar(nc, "time", eltype, ("time",))
+    time.attrib["standard_name"] = standard_name
+    time.attrib["long_name"] = long_name
+    time.attrib["axis"] = axis
+    for (k, v) in kwargs
+        time.attrib[String(k)] = v
+    end
+    return time
+end
+
+
+"""
+    def_space_coord(nc::NCDataset, space::Spaces.AbstractSpace; type = "cgll")
+
+Add spatial dimensions for `space` in the NetCDF dataset `nc`, compatible with
+the type used by [`remap_weights`](@ref).
+
+If a compatible dimension already exists, it will be reused.
+"""
+function def_space_coord(
+    nc::NCDataset,
+    space::Spaces.SpectralElementSpace2D;
+    type = "cgll",
+)
+    if type == "cgll"
+        nodes = Spaces.unique_nodes(space)
+        ncol = length(nodes)
+    else
+        error("Unsupported type: $type")
+    end
+
+    if haskey(nc, "lon")
+        # dimension already exists: check correct size
+        if size(nc["lon"]) != (ncol,)
+            error("incompatible horizontal dimension already exists")
+        end
+        return nothing
+    end
+
+    # dimensions
+    defDim(nc, "ncol", ncol)
+
+    # variables
+    ## lon
+    lon = defVar(nc, "lon", Float64, ("ncol",))
+    lon.attrib["units"] = "degrees_east"
+    lon.attrib["axis"] = "X"
+    lon.attrib["long_name"] = "longitude"
+    lon.attrib["standard_name"] = "longitude"
+
+    ## lat
+    lat = defVar(nc, "lat", Float64, ("ncol",))
+    lat.attrib["units"] = "degrees_north"
+    lat.attrib["axis"] = "Y"
+    lat.attrib["long_name"] = "latitude"
+    lat.attrib["standard_name"] = "latitude"
+
+    coords = Spaces.coordinates_data(space)
+
+    for (col, ((i, j), e)) in enumerate(nodes)
+        coord = slab(coords, e)[i, j]
+        lon[col] = coord.long
+        lat[col] = coord.lat
+    end
+    nc.attrib["node_type"] = type
+    return lat, lon
+end
+
+function def_space_coord(
+    nc::NCDataset,
+    space::Spaces.CenterFiniteDifferenceSpace,
```

Comment:
I don't mean to slow things down but, should we also support `FaceFiniteDifferenceSpace `? If so, we'll need to have `zf` / `zc` defined variables, and maybe a helper to get the appropriate coordinate name

Reply:
I suppose vertical velocity is one variable that someone might want

Reply:
Or we could write a wrapper that interpolates to a cell center field.

Reply:
Ah, this may not be needed if TempestRemap is collocated at cell centers.

Reply:
TempestRemqp doesn't do anything to the vertical.

I'm not sure what we should call the face locations (`z_half`?). @szy21 is there a standard name that is used?

Reply:
`z_half` seems to be a common name (I'm not sure if it's standard). I think as long as we all agree that half levels are on cell faces then it's fine.

Reply:
`z_half` seems to be a common name (I'm not sure if it's standard). I think as long as we all agree that half levels are on cell faces then it's fine.

Reply:
okay, i went with that.